src/main/java/net/goldiriath/plugin/game/mobspawn/MobSpawnProfile.java

```diff
@@ -150,30 +148,81 @@ public NPC spawn(Location location) {
         return npc;
     }
 
-    public boolean hasSpawnThreshold() {
-        return spawnThreshold >= 0;
-    }
-
     @Override
     public void loadFrom(ConfigurationSection config) {
 
+        // Meta
+        customName = config.getString("name", null);
+        if (customName != null) {
+            customName = ChatColor.translateAlternateColorCodes('&', customName);
+        }
+        timeThreshold = config.getInt("spawn_threshold", -1);
+
         final String entityTypeName = config.getString("type", null);
         if (entityTypeName == null) {
             type = null;
         } else {
-            type = EntityType.fromName(entityTypeName);
+            try {
+                type = EntityType.valueOf(entityTypeName);
+            } catch (IllegalArgumentException ex) {
+                type = null;
+                logger.warning("Could not load mobspawn profile '" + id + "'. Invalid entity type: " + entityTypeName);
+            }
+        }
+
+        // Stats
+        level = config.getInt("stats.level", -1);
+        if (level == -1) {
+            logger.warning("Could not load mobspawn profile '" + id + "'. Level field not present.");
+        }
+        damage = config.getInt("stats.damage", -1);
+        health = config.getInt("stats.health", -1);
+        if (health == -1) {
+            logger.warning("Could not load mobspawn profile '" + id + "'. Health field not present.");
```

Comment:
@cashenkes The mob is not spawned. The `loadFrom` method always succeeds. However, this does not mean that the mobspawn profile is valid. There is an `isValid` method which is called before the mobspawn is added. See https://github.com/Goldiriath/Goldiriath/pull/68/files/611d9d1e81454a082c34a9f3ac9a4cd5a6038676#diff-21574682e7b72292cb3ce93e74dc21aaR360 If the health value is not set properly, the `isValid` method returns false and the mobspawn profile is not added to the list of currently loaded profiles.